### PR TITLE
Enable use of CMake in github actions .yml

### PR DESCRIPTION
**Pull request by gionellef**
## Description

Creates a "build" directory outside the project (not in source)
Uses existing CMakeLists.txt, which includes CTest driver, for compilation and testing.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -18,6 +18,15 @@ jobs:
...
             OCCA_COVERAGE: 1
             OCCA_FORTRAN_ENABLED: 1
+            useCMake: false
+
+          - name: (Ubuntu) CMake + gcc-8
+            os: ubuntu-18.04
+            CC: gcc-8
+            CXX: g++-8
+            OCCA_COVERAGE: 1
+            OCCA_FORTRAN_ENABLED: 1
```
Comment (awehrfritz) [inline .github/workflows/build.yml:28]: The Fortran parts don't include CMake files, neither the library nor the examples/tests. I suggest to disable the Fortran interface in the CMake runs for the time being.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -18,6 +18,15 @@ jobs:
             FC: gfortran-8
             OCCA_COVERAGE: 1
             OCCA_FORTRAN_ENABLED: 1
+            useCMake: false
```
Comment (awehrfritz) [inline .github/workflows/build.yml:21]: This line is unnecessary and can be removed.

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -81,4 +107,4 @@ jobs:
       run: ./tests/run_examples
 
     - name: Upload code coverage
-      uses: codecov/codecov-action@v1
+      uses: codecov/codecov-action@v1
```
Comment (awehrfritz) [inline .github/workflows/build.yml:110]: There needs to be an empty line at the end of the file.

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -70,9 +79,26 @@ jobs:
 
     - name: Compile library
       run: make -j
+      if: ${{ !matrix.useCMake }}
```
Comment (awehrfritz) [inline .github/workflows/build.yml:82]: Please add this line also to the `Compiler info` step. Printing the make settings for a CMake build doesn't make much sense.

**Comment by dmed256** [comment]
@gionellef Thanks for making all of these changes! Looks like the failures are from warnings that are fixed in #329. I think we can merge this soon though, excited about having CMake as part of the test suite :)

**Comment by dmed256** [comment]
@gionellef Looks like the tests are running and passing! However, there seems to be an issue when running the example codes from the CMake build directory

**Comment by gionellef** [comment]
Oh wow I didn’t realize the problem could be that simple :D it was my pleasure, thanks a lot for letting me learn!!

**Comment by dmed256** [comment]
@gionellef will you have time to finish this PR?

**Pull request closed without merging**